gqlalchemy/transformations/translators/nx_translator.py

```diff
@@ -0,0 +1,279 @@
+    pass
+
+
+class NxTranslator(Translator):
+    """Uses original ids from Memgraph. Labels are encoded as properties. Since Networkx allows
+    that nodes have properties of different dimensionality, this modules makes use of it and stores properties
+    as dictionary entries. All properties are saved to Networkx data structure.
+    """
+
+    def __init__(self, default_node_label="NODE", default_edge_type="RELATIONSHIP",
+        host: str = MG_HOST,
+        port: int = MG_PORT,
+        username: str = MG_USERNAME,
+        password: str = MG_PASSWORD,
+        encrypted: bool = MG_ENCRYPTED,
+        client_name: str = MG_CLIENT_NAME,
+        lazy: bool = MG_LAZY,) -> None:
+            self.__all__ = ("nx_to_cypher", "nx_graph_to_memgraph_parallel")
+            super().__init__(default_node_label, default_edge_type, host, port, username, password, encrypted, client_name, lazy)
+
+
+    def to_cypher_queries(self, graph: nx.Graph, config: NetworkXCypherConfig = None) -> Iterator[str]:
+        """Generates a Cypher query for creating a graph."""
+
+        if config is None:
+            config = NetworkXCypherConfig()
+
+        builder = NetworkXCypherBuilder(config=config)
+
+        yield from builder.yield_queries(graph)
+
+    def nx_graph_to_memgraph_parallel(
+        self,
+        graph: nx.Graph,
+        host: str = "127.0.0.1",
+        port: int = 7687,
+        username: str = "",
+        password: str = "",
+        encrypted: bool = False,
+        config: NetworkXCypherConfig = None,
+    ) -> None:
+        """Generates Cypher queries and inserts data into Memgraph in parallel."""
+        if config is None:
+            config = NetworkXCypherConfig()
+
+        builder = NetworkXCypherBuilder(config=config)
+        query_groups = builder.yield_query_groups(graph)
+
+        if not config.create_index:
+            self._check_for_index_hint(
+                host,
+                port,
+                username,
+                password,
+                encrypted,
+            )
+
+        for query_group in query_groups:
+            self._start_parallel_execution(query_group, host, port, username, password, encrypted)
+
+    def _start_parallel_execution(
+        self, queries_gen: Iterator[str], host: str, port: int, username: str, password: str, encrypted: bool
+    ) -> None:
+        num_of_processes = mp.cpu_count() // 2
+        queries = list(queries_gen)
+        chunk_size = len(queries) // num_of_processes
+        processes = []
+
+        for i in range(num_of_processes):
+            process_queries = queries[i * chunk_size : chunk_size * (i + 1)]
+            processes.append(
+                mp.Process(
+                    target=self._insert_queries,
+                    args=(
+                        process_queries,
+                        host,
+                        port,
+                        username,
+                        password,
```

Comment:
And why are we here sending again host, port, username and password, don't we have it in init?

Reply:
Yes same as above, it is old method.